object.js

```diff
@@ -149,22 +191,76 @@ function hint(gj, options) {
         }
     }
 
-    function crs(_) {
-        if (!_.crs) return;
-        if (typeof _.crs === 'object') {
-            var strErr = requiredProperty(_.crs, 'type', 'string'),
-                propErr = requiredProperty(_.crs, 'properties', 'object');
-            if (!strErr && !propErr) {
-                // http://geojson.org/geojson-spec.html#named-crs
-                if (_.crs.type === 'name') {
-                    requiredProperty(_.crs.properties, 'name', 'string');
-                } else if (_.crs.type === 'link') {
-                    requiredProperty(_.crs.properties, 'href', 'string');
+
+    function rightHandRule (geometry) {
+        var rhr = true;
+        if (geometry.type == 'Polygon') {
+            rhr = isPolyRHR(geometry.coordinates);
+        } else if (geometry.type == 'MultiPolygon') {
+            for (var i = 0; i < geometry.coordinates.length; i++) {
+                if (!isPolyRHR(geometry.coordinates[i])) {
+                    rhr = false;
+                    break;
                 }
             }
+        }
+        if (!rhr) {
+            errors.push({
+                message: 'Polygons and MultiPolygons should follow the right-hand rule',
+                line: geometry.__line__
+            });
+        }
+    };
+
+    function isPolyRHR (coords) {
+        if (coords && coords.length > 0) {
+            if (!isRingClockwise(coords[0]))
+                return false;
+            for (var i = 1; i < coords.length; i++) {
+                if (isRingClockwise(coords[i]))
+                    return false;
+            }
+        }
+        return true;
+    };
+
+    function isRingClockwise (coords) {
+        var area = 0;
+        if (coords.length > 2) {
+            var p1, p2;
+            for (var i = 0; i < coords.length - 1; i++) {
+                p1 = coords[i];
+                p2 = coords[i + 1];
+                area += rad(p2[0] - p1[0]) * (2 + Math.sin(rad(p1[1])) + Math.sin(rad(p2[1])));
+            }
+        }
+
+        if (area >= 0)
+            return true;
+        else
+            return false;
+    };
+
+    function rad(x) {
+        return x * Math.PI / 180;
+    }
+
+    function crs(_) {
+        if (!_.crs) return;
+        var defaultCRS = {
+            "type": "name",
+            "properties": {
+              "name": "urn:ogc:def:crs:OGC:1.3:CRS84"
+              }
+        }
+        if (typeof _.crs === 'object' && _.crs.properties && _.crs.properties.name == defaultCRS.properties.name) {
```

Comment:
given that javascript doesn't do object equivalence, it'd be simpler to store only the default crs name as a variable rather than the object